lib/matplotlib/backend_bases.py

```diff
@@ -1553,6 +1553,20 @@ def _draw(renderer): raise Done(renderer)
             return figure._cachedRenderer
 
 
+def is_non_interactive_terminal_ipython(ip):
```

Comment:
This does only check attributes of `ip`, so I feel it should belong to the IPython object and should be implemented there as `IPython.is_interactive` or similar.

Then one can do:

~~~
if ip.is_interactive:
    ip.enable_matplotlib()
~~~

Or maybe even in `enable_matplotlib()`

~~~
def enable_matplotlib(self):
    if not self.is_interactive:
        warnings.warn('enable_matplotlib() called on a non-interactive IPython.'
                      'Matplotlib interactive mode is not enabled.'
                      'This will raise a RuntimeError in furture versions.')
        return
    ...
~~~

Reply:
But then we would only support versions of ipython>=whatever-implements-this.

Reply:
Which is the reason I did it this way, and also `ip.is_interactive` make no-sens on non-`TerminalInteractiveShell`. When running matplotlib in notebook, you will get a different type for `ip` (ZmqInteractiveShell). And both are subclass of a base class which can be overridden by other implementations.

To be clearer this function test:
1) is ip created by a **terminal** IPython. 
2) if 1; is it interactive.

Reply:
I'm ok with the present approach for back-comatibility.